_example/servers/sp-server/sp-server.go

```diff
@@ -112,18 +111,18 @@ func main() {
 		},
 	}
 
-	middleware := sp.NewMiddleware(&serviceProvider)
-
 	r := chi.NewRouter()
 	r.Use(logHandler)
 
-	r.Get(metadataPath, middleware.ServeMetadata)
-	r.Post(acsPath, middleware.ServeAcs(accessGranted))
+	r.Get(metadataPath, serviceProvider.MetadataHandler)
+
+	r.With(serviceProvider.AssertionConsumer).
+		Post(acsPath, accessGrantedHandler)
 
 	log.Printf("Test SP server listening at %s (%s)", *flagListenAddr, *flagPublicURL)
 	switch *flagInitiatedBy {
 	case "sp":
-		r.Get("/", setRelayState(middleware.ServeRequestAuth))
+		r.Get("/", setRelayState(serviceProvider.RequestAuthHandler))
```

Comment:
How do they call the SP-initiated login in a SAML spec? AuthNRequest? It'd be probably a good idea to stick with their naming conventions as much as possible.. maybe `sp.AuthNRequestHandler`? On the other hand, this looks good too

Reply:
What do you think?

Let's try to make the interfaces/func names as simple to understand as possible to all pkg consumers :)

Reply:
Good call! I like it